src/components/tl/ucp/tl_ucp_coll.h

```diff
@@ -208,18 +208,35 @@ ucc_tl_ucp_init_task(ucc_base_coll_args_t *coll_args, ucc_base_team_t *team)
     ucc_tl_ucp_team_t *tl_team = ucc_derived_of(team, ucc_tl_ucp_team_t);
     ucc_tl_ucp_task_t *task    = ucc_tl_ucp_get_task(tl_team);
 
+    if (ucc_unlikely(!task)) {
+        return NULL;
+    }
+
     ucc_coll_task_init(&task->super, coll_args, team);
     if (coll_args->mask & UCC_COLL_ARGS_FIELD_TAG) {
-        task->tagged.tag           = task->tagged.tag;
-    } else if (coll_args->args.coll_type == UCC_COLL_TYPE_SEND ||
-               coll_args->args.coll_type == UCC_COLL_TYPE_RECV) {
-        /* Use the maximum tag with the point-to-point bit set as a reserved tag
-         * for point-to-point messages that do not otherwies have a tag set. */
-        task->tagged.tag           = UCC_TL_UCP_MAX_TAG;
+        task->tagged.tag = coll_args->args.tag;
+    } else if (UCC_COLL_ARGS_ACTIVE_SET(&coll_args->args)) {
```

Comment:
Shouldn't this be an `if` instead of an `else if` since you can have both `UCC_COLL_ARGS_FIELD_TAG` on and use an active set at the same time?

Or are we saying that you can't use both? I see from the code in `src/components/tl/ucp/tl_ucp_tag.h` below that `UCC_TL_UCP_ACTIVE_SET_2_TAG` is set to a specific tag instead of a specific bit.

Reply:
you are right.